### PR TITLE
Sm 49 graph ql user crud

### PR DESCRIPTION
**Pull request by nivrutti1234**

### CODE REVIEW

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/impl/DefaultDesignationService.java [hunk 1]
```diff
@@ -57,8 +57,8 @@ public void delete(Designation designation) {
 
 	@Override
 	public List<Designation> findAll() {
-		List<Designation> result = repository.findAll();
-		return result;
+		Optional<List<Designation>> result = repository.findByIsActive(true);
```
Comment (ankit11088) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/service/impl/DefaultDesignationService.java:60]: Again don't hardcode.

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -110,13 +122,16 @@ public UserResponseDto findById(String id) {
...
     @Override
-    public void delete(String userId) {
+    public Status delete(String userId) {
 
         Optional<User> optUser = userRepository.findById(userId);
         if (optUser.isPresent()) {
             User user = optUser.get();
             user.setActive(false);
             userRepository.save(user);
+            return Status.SUCCESS;
```
Comment (ankit11088) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/service/impl/UserServiceImpl.java:132]: Either handle the exception via try-catch and try can say SUCCESS and catch can throw Exception.
If not, then don't assume..Status.SUCCESS;

File: ClariseMain/src/main/java/com/clairvoyant/clarise/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -131,27 +146,18 @@ public List<UserResponseDto> findAll() {
             UserResponseDto userResponseDto = new UserResponseDto();
             BeanUtils.copyProperties(user, userResponseDto);
 
-            Designation designation = new Designation();
-            for (UserDesignationMapping userDesignationMapping:userDesignationMappings) {
-                if (user.getId().equals(userDesignationMapping.getUser().getId())){
-                    designation = userDesignationMapping.getDesignation();
-                }
-            }
+            Designation designation = userDesignationMappings.stream()
```
Comment (ankit11088) [inline ClariseMain/src/main/java/com/clairvoyant/clarise/service/impl/UserServiceImpl.java:149]: findAll() for users
Get All Users > UserDesignation
			> UserRole
			> UserCategory


User {

	@OneToMany
	UserRole

	@OneToOne
	UserDesignation

	@OneToMany
	UserCategory
}

Comment (ankit11088) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/service/impl/UserServiceImpl.java:149]: User {

	@OneToMany
	Set<UserRole>

	@OneToOne(FetchType.LAZY)
	UserDesignation

	@OneToMany
	Set<UserCategory> ;
}



UserCategory {
	@ManyToOne
	User;
}

UserRole {
	@ManyToOne
	User;
}

**Review by ankit11088 — CHANGES_REQUESTED** [review]
Address the comments mentioned.

File: ClariseMain/src/main/java/com/clairvoyant/clarise/controller/UserController.java [hunk 4]
```diff
@@ -42,19 +42,18 @@ public UserResponseDto getUserById(@PathVariable String userId)
     }
 
     @PostMapping
-    public User addOrUpdateUser(@RequestBody UserDto userDto) {
+    public UserResponseDto addOrUpdateUser(@RequestBody UserDto userDto) {
```
[the remark this answers is not in the document]
Comment (nivrutti1234) [reply ClariseMain/src/main/java/com/clairvoyant/clarise/controller/UserController.java:45]: I have made changes in the UserService, so it is required to make change in the controller.

**Review by ankit11088 — APPROVED** [review]
Approving based on what we agreed upon.

**Pull request merged**